### PR TITLE
kvstore: add support for reading/writing transaction to checkpoint metadata

### PR DESCRIPTION
**Pull request by bmwill**
## Description 

Adds support for reading/writing transaction to checkpoint metadata to kvstore. Please check that i've properly covered all the necessary pieces since this is my first time modifying this system.

## Test Plan 

How did you test the new or updated feature?

---
If your changes are not user-facing and not a breaking change, you can skip the following section. Otherwise, please indicate what changed, and then add to the Release Notes section as highlighted during the release process.

### Type of Change (Check all that apply)

- [ ] protocol change
- [ ] user-visible impact
- [ ] breaking change for a client SDKs
- [ ] breaking change for FNs (FN binary must upgrade)
- [ ] breaking change for validators or node operators (must upgrade binaries)
- [ ] breaking change for on-chain data layout
- [ ] necessitate either a data wipe or data migration

### CODE REVIEW

**Comment by bmwill** [comment]
Since this metadata is only ever read once at a time (ie reads for this are not batched in RPC today) i added a non-multi-get function for this data since it seemed much less complex than adding mult-get support and then not using it.

**Comment by phoenix-o** [comment]
we'd need a similar PR to https://github.com/MystenLabs/sui/pull/13568 for the initial backfill with a different progress key or combine both

File: crates/sui-json-rpc/src/read_api.rs [hunk 1]
```diff
@@ -727,20 +727,14 @@ impl ReadApiServer for ReadApi {
...
-                    .map_err(|e| {
-                        error!("Failed to retrieve checkpoint sequence for transaction {digest:?} with error: {e:?}");
-                        Error::from(e)
-                    })
-            }).await.map_err(Error::from)?? {
-                temp_response.checkpoint_seq = Some(seq);
-            }
+            temp_response.checkpoint_seq = self
+                .transaction_kv_store
+                .deprecated_get_transaction_checkpoint(digest)
```
Comment (mystenmark) [inline crates/sui-json-rpc/src/read_api.rs:732]: are we sure this isn't called in a loop? if so we will need multi-get or the latency will be awful

Comment (bmwill) [reply crates/sui-json-rpc/src/read_api.rs:732]: yes, afaik this is a single jsonrpc request. fwiw the checkpoint querying below is done similarly

**Pull request merged**